test/runnable/testbounds.d

```diff
@@ -185,34 +185,53 @@ void test13976()
 {
     int[] da = new int[](10);
     int[10] sa;
-    size_t l = 0;               // upperInRange
-    size_t u = 9;               // | lowerLessThan
-                                // | |  check code
-    { auto s = da[l .. u];   }  // 0 0  (u <= 10 && l <= u  )
-    { auto s = da[1 .. u];   }  // 0 0  (u <= 10 && l <= u  )
-    { auto s = da[l .. 10];  }  // 0 0  (u <= 10 && l <= u  )
-    { auto s = da[1 .. u%5]; }  // 0 0  (u <= 10 && l <= u%5)
-
-    { auto s = da[l .. u];   }  // 0 0  (u   <= 10 && l <= u)
-    { auto s = da[0 .. u];   }  // 0 1  (u   <= 10          )
-    { auto s = da[l .. 10];  }  // 0 0  (u   <= 10 && l <= u)
-    { auto s = da[0 .. u%5]; }  // 0 1  (u%5 <= 10          )
-
-    { auto s = sa[l .. u];   }  // 0 0  (u <= 10 && l <= u  )
-    { auto s = sa[1 .. u];   }  // 0 0  (u <= 10 && l <= u  )
-    { auto s = sa[l .. 10];  }  // 1 0  (           l <= u  )
-    { auto s = sa[1 .. u%5]; }  // 1 0  (           l <= u%5)
-
-    { auto s = sa[l .. u];   }  // 0 0  (u <= 10 && l <= u )
-    { auto s = sa[0 .. u];   }  // 0 1  (u <= 10           )
-    { auto s = sa[l .. 10];  }  // 1 0  (           l <= 10)
-    { auto s = sa[0 .. u%5]; }  // 1 1  NULL
+    enum size_t two = 2;
+    enum size_t five = 5;
+    size_t lb = 0;                    // upperInRange
+    size_t ub = 9;                    // | lowerLessThan
+                                      // | |  check code
+    { auto s = da[lb .. ub];        } // 0 0  (ub   <= $  && lb <= ub  )
+    { auto s = da[1 .. ub];         } // 0 0  (ub   <= $  && 1  <= ub  )
+    { auto s = da[lb .. 10];        } // 0 0  (10   <= $  && lb <= 10  )
+    { auto s = da[1 .. ub%5];       } // 0 0  (ub%5 <= $  && 1  <= ub%5)
+
+    { auto s = da[lb .. ub];        } // 0 0  (ub   <= $  && lb <= ub  )
+    { auto s = da[0 .. ub];         } // 0 1  (ub   <= $               )
+    { auto s = da[lb .. 10];        } // 0 0  (10   <= $  && lb <= 10  )
+    { auto s = da[0 .. ub%5];       } // 0 1  (ub%5 <= $               )
+
+    { auto s = da[0 .. 0];          } // 0 1  (0    <= $               )
+    { auto s = da[0 .. $];          } // 0 1  ($    <= $               )
```

Comment:
These two can be improved in a latter PR by using the check for `0` or `$` from #7677.